### PR TITLE
Check if the topic arn exists before publish

### LINKED ISSUE #1319: aws sns publish against non existent topic does not throw exception
With latest localstack image running in Docker, when publishing a message to a non-existent SNS topic no error is thrown.  This differs from AWS.

### PR DESCRIPTION
**Pull request by ImFlog**
This PR removes the auto registering of topic when receiving requests and throw an error when a publish is called on a non existent topic (just like SNS does).
I also tried this with a Java client and it works :)

Fixes #1319.

### CODE REVIEW

**Comment by whummer** [comment]
Thanks for this change. Looks like we still have an error in the CI builds:
```
localstack.services.generic_proxy: ERROR: Error forwarding request: 'NoneType' object has no attribute 'append' Traceback (most recent call last):
  File "/home/travis/build/localstack/localstack/localstack/services/generic_proxy.py", line 241, in forward
    updated_response = self.proxy.update_listener.return_response(**kwargs)
  File "/home/travis/build/localstack/localstack/localstack/services/sns/sns_listener.py", line 93, in return_response
    attributes
  File "/home/travis/build/localstack/localstack/localstack/services/sns/sns_listener.py", line 168, in do_subscribe
    SNS_SUBSCRIPTIONS.get(topic_arn).append(subscription)
AttributeError: 'NoneType' object has no attribute 'append'
```

File: tests/integration/test_sns.py [hunk 1]
```diff
@@ -72,3 +74,14 @@ def test_attribute_raw_subscribe(self):
...
         msg_received = msgs['Messages'][0]
         self.assertEqual(message, msg_received['Body'])
+
+    def test_unknown_topic_publish(self):
+        sns_client = aws_stack.connect_to_service('sns')
+        fake_arn = 'arn:aws:sns:us-east-1:123456789012:i_dont_exist'
+        message = u'This is a test message'
+        try:
+            sns_client.publish(TopicArn=fake_arn, Message=message)
+        except ClientError as e:
```
Comment (whummer) [inline tests/integration/test_sns.py]: We should raise an assertion error if the previous call on line 83 passes.

**Comment by ImFlog** [comment]
The PR is here because of this create_topic that happens everytime.
I added a CreateTopic handler to replace it.

I'm blind about the impact of my changes because I can't find a way to run the all the tests locally. Lots of them always fail (I'm on MacOS). So I just test the piece of code I change ...
Anything I can do to run them locally ?

**Comment by efritz** [comment]
> Anything I can do to run them locally ?

I run the tests via docker with `make docker-build`. This is essentially what Travis does.

**Comment by ImFlog** [comment]
@efritz thank you for the tips. It's a bit better but I still have issue with what's supposed to run on 4564 (Dynamo ?).

@whummer I understood what was wrong and added the do_create in the `return_response` where I have access to the topic_arn. Now it runs well (tested create and publish methods with a Java client) and the test works for mostly everything but two non related tests seems to fail (in python3).
Are they flaky tests ?

Errors:
```
======================================================================
FAIL: test_api_gateway_lambda_proxy_integration_any_method (tests.integration.test_api_gateway.TestAPIGatewayIntegrations)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/localstack/localstack/tests/integration/test_api_gateway.py", line 210, in test_api_gateway_lambda_proxy_integration_any_method
    self.API_PATH_LAMBDA_PROXY_BACKEND_ANY_METHOD)
  File "/home/travis/build/localstack/localstack/tests/integration/test_api_gateway.py", line 254, in _test_api_gateway_lambda_proxy_integration_any_method
    self.assertEqual(result.status_code, 200)
AssertionError: 500 != 200
```
```
=====================================================================
FAIL: test_destroy_idle_containers (tests.integration.test_lambda.TestDockerBehaviour)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/localstack/localstack/tests/integration/test_lambda.py", line 644, in test_destroy_idle_containers
    self.assertEqual(len(executor.get_all_container_names()), 1)
AssertionError: 0 != 1
```

**Comment by whummer** [comment]
Tests are now fixed on `master` - thanks again. 👍

**Pull request merged**